lib/rubocop/cop/lint/useless_constant_scoping.rb

```diff
@@ -56,7 +56,11 @@ def on_casgn(node)
         private
 
         def after_private_modifier?(left_siblings)
-          left_siblings.compact.select(&:send_type?).any? { |node| node.command?(:private) }
+          access_modifier_candidates = left_siblings.compact.select do |left_sibling|
+            left_sibling.respond_to?(:send_type?) && left_sibling.send_type?
```

Comment:
This fixes the exception, but it still ends up with a false negative for case reported in the issue:

```ruby
class Foo
  private
  ENV['RAILS_ENV'] = RAILS_ENV = Rails.env.to_s
end
```
(this won't flag an offense since the left siblings of the `RAILS_ENV` assignment node don't contain the `private` modifier). I think to detect the `private` modifier, you'd have to walk up the tree while the parent is an `assignment?` node and then check _its_ left siblings.

Reply:
I see. Whether this corner use case should be registered as an offense is a bit debatable. For now, this PR focuses on resolving the error, and I will handle this issue separately. Thank you for the feedback.